nearcore/src/lib.rs

```diff
@@ -423,7 +429,9 @@ pub fn recompress_storage(home_dir: &Path, opts: RecompressOpts) -> anyhow::Resu
         None
     };
 
-    let dst_store = dst_opener.open();
+    let dst_store = dst_opener
+        .open()
+        .with_context(|| format!("Opening database at {}", dst_opener.path().display()))?;
```

Comment:
Yeah, I think you can add `path` to message in the openner -- this context doesn't add anything from the call-site, so it could be in the callee.

Reply:
The reason for doing it here is that it’s just a single location.  In the callee there are multiple exit paths and each of them would need to be changed.

Reply:
Uh, I actually intended to wait for your reply, but automerge was still on.  I can still change it in another PR.

Reply:
>The reason for doing it here is that it’s just a single location.

But we are doing this 4 times at the call-site, while there are 2 exit paths at callee :) 

But yeah, this is annoying. In the ideal world we'd have `try` blocks. Today, I use a mix of (in this order)

* just manually contexing every exit point
* adding a `try_foo` private helper function
* immediately-invoked lambda. 

No strong opinion though!